Jint/Engine.cs

```diff
@@ -292,6 +292,8 @@ public Engine Execute(Script script)
         /// </summary>
         private Engine ScriptEvaluation(ScriptRecord scriptRecord)
         {
+            DebugHandler.OnParsed(scriptRecord.EcmaScriptCode);
```

Comment:
nit: script might be existing `Script` instance and thus not parsed in literal sense

Reply:
Well it's internal so it's all about splitting hairs here. `Before`/`After` is good pairing, maybe even `OnEvaluating`/`OnEvaluated`, but all good as it's again, internal 🙂

Reply:
Well, the event that it ends up triggering - `Parsed` - isn't internal. 🙂

Reply:
Ah true. We're still in beta, we can break things even after this 😉

Reply:
But the Debugger API has been so constantly breaking that I'd like for once to make a nice choice. 😐 Ended up with `BeforeEvaluate` (never liked `-ing` in event names, because it can both indicate "before" and "during").

And, more importantly, moved it into `EvaluateModule` rather than `LoadModule`, which is also used for loading modules for other purposes than evaluation. If that looks OK, it's good to go, I think.

Reply:
Yeah, because now it doesn't take `LinkModule` into account, only `EvaluateModule` - and `LinkModule` is actually the main reason it's needed (scripts you don't pass to the engine yourself). 😋 But `LinkModule` doesn't do evaluation, so the event name, once again, becomes meaningless... 🤨

Reply:
So there... The name stays for now, although it's now moved back to `LoadModule` 😋 Added a better test case (for two levels of import), and also tested again with Jint.ExampleDebugger and Jint.DebugAdapter.